### DISCUSSION TITLE
MongoDB support isn't compiled (errno 22, Invalid argument)

### DISCUSSION THREAD

**Discussion by jackcongx** [netdata/netdata#12713] [Q&A]
![image_1](https://user-images.githubusercontent.com/52734448/163938287-72cdedec-37e6-4c69-9235-1eb86fe4785d.png)

**Comment by MrZammler** [comment]
Hi @jackcongx . Can you share a bit more about this? How did you install Netdata, and also have you checked [Export metrics to MongoDB](https://learn.netdata.cloud/docs/agent/exporting/mongodb/) ?

**Reply by jackcongx** [reply]
I use `bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)` to install netdata
![image_2](https://user-images.githubusercontent.com/52734448/163947666-d4368d39-58ca-4eb3-9e66-13ec8f117caa.png)

**Comment by MrZammler** [comment]
Thanks! Do you have `libmongoc` installed?

**Reply by jackcongx** [reply]
yes,first i instaled mongo-c-driver1.21.1,later  i instaled mongo-c-driver1.16.2

**Reply by MrZammler** [reply]
Ok, thank you. So most likely our static build does not contain the required code. We'll check.

In the mean time, it should detect it if you compile from [source](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/kickstart?_escaped_fragment_=) if it's possible.

**Reply by jackcongx** [reply]
thank you, i will try it

**Reply by jackcongx** [reply]
Hi, @MrZammler ,Can you help me,How to re-install Netdata from the source?

**Reply by vlvkobal** [reply]
There is a magic script - `netdata-installer.sh` 😃. Please read the [documentation](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/manual).

**Reply by jackcongx** [reply]
I use `netdata-installer.sh`,but something went wrong
![f469cb2b900dce959269609920c918a](https://user-images.githubusercontent.com/52734448/164184217-710d98bd-2d7f-44d8-baad-017ece7dcf9c.png)

**Reply by vkalintiris** [reply]
I think you need to fetch the source code of the git submodules first by running:

```
git submodule update  --init --recursive --force
```

**Reply by Ferroin** [reply]
Assuming you had an existing static install, the following should reinstall with a source build:

```sh
wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --reinstall-clean --build-only
```
